lib/chef/knife/clc_server_create.rb

```diff
@@ -431,8 +440,14 @@ def prepare_launch_params
       end
 
       def prepare_ip_params
+        ports = []
+        ports << config[:clc_allowed_protocols]
+        if config[:bootstrap_protocol]
+          ports << { 'protocol' => 'tcp', 'port' => 5985 }
```

Comment:
This just won't work. What if protocol is SSH? Should we still open port 5985? I think we should not assign IP by ourselves at all. By doing this, you restrict user from bootstrapping from private network. Take a look how it's done for Linux host (watch how `bootstrap_private` is used).